### PR TITLE
[c/fr] C French translation

### PR DESCRIPTION
**Pull request by sun-exploit**
New C French translation c-fr.html.markdown file

- [x] I solemnly swear that this is all original content of which I am the original author
- [x] Pull request title is prepended with `[language/lang-code]`
- [x] Pull request touches only one file (or a set of logically related files with similar changes made)
- [x] Content changes are aimed at *intermediate to experienced programmers* (this is a poor format for explaining fundamental programming concepts)
- [x] If you've changed any part of the YAML Frontmatter, make sure it is formatted according to [CONTRIBUTING.md](https://github.com/adambard/learnxinyminutes-docs/blob/master/CONTRIBUTING.markdown)
  - [x] Yes, I have double-checked quotes and field names!

### CODE REVIEW

**Comment by zethis** [comment]
Nice to have !

File: fr-fr/c-fr.html.markdown [hunk 1]
```diff
@@ -0,0 +1,863 @@
...
+  unsigned short ux_short;
+  unsigned int ux_int;
+  unsigned long long ux_long_long;
+
+  // Les caractères entre guillemets simples sont des entiers d'un jeu de caractères machine.
+  '0'; // => 48 dans le jeu de caractères ASCII.
+  'A'; // => 65 dans le jeu de caractères ASCII.
+
+  // sizeof(T) vous rend la taille d'une variable de type T en octets.
+  // sizeof(obj) revoie la taille en sortie de l'expression (variable, literal, etc.).
```
Comment (vendethiel) [inline fr-fr/c-fr.html.markdown:138]: "literal"

la version non-type est `sizeof obj`

File: fr-fr/c-fr.html.markdown [hunk 2]
```diff
@@ -0,0 +1,863 @@
...
+  printf("%zu\n", sizeof(int)); // => 4 (sur la plupart des machines les mots font 4 octets)
+
+  // Si l'argument de l'opérateur `sizeof` est une expression, alors son argument
```
Comment (vendethiel) [inline fr-fr/c-fr.html.markdown:141]: à la différence de quel autre cas ... ?

File: fr-fr/c-fr.html.markdown [hunk 3]
```diff
@@ -0,0 +1,863 @@
...
+
+/*
+Les fonctions sont appelées par valeur. Lorsqu'une fonction est appelée, les arguments passés à
+à la fonction sont une copie des arguments originaux (sauf pour les tableaux). Tout ce que vous
+allez faire aux arguments de la fonction ne changera pas la valeur des arguments
+originaux avant appel de la fonction.
+
+Utilisez les pointeurs si vous avez besoin de modifier la valeur d'origine des arguments.
+
+Exemple: inversion sur place d'une chaîne de caractères
```
Comment (vendethiel) [inline fr-fr/c-fr.html.markdown:557]: espace avant `:`

File: fr-fr/c-fr.html.markdown [hunk 4]
```diff
@@ -0,0 +1,863 @@
...
+*/
+
+// Une fonction void retourne aucune valeur
```
Comment (vendethiel) [inline fr-fr/c-fr.html.markdown:560]: *ne retourne

File: fr-fr/c-fr.html.markdown [hunk 5]
```diff
@@ -0,0 +1,863 @@
...
+int areaptr(const rect *r)
+{
+  return r->width * r->height;
+}
+
+///////////////////////////////////////
+// Pointeurs sur fonction
+///////////////////////////////////////
+/*
+At run time, functions are located at known memory addresses. Function pointers are
```
Comment (vendethiel) [inline fr-fr/c-fr.html.markdown:701]: (La version non traduite n'a pas été supprimée)

File: fr-fr/c-fr.html.markdown [hunk 6]
```diff
@@ -0,0 +1,863 @@
...
+/* un fichier d'en-tête mais placées plutôt dans des en-têtes séparés ou un fichier C.                    */
+
+#endif /* Fin de la directive if du precompilateur. */
+
+```
+## Lectures complémentaires
+
+Le mieux est de vous trouver une copie de [K&R, aka "The C Programming Language"](https://en.wikipedia.org/wiki/The_C_Programming_Language)
+C'est *le* livre sur C, écrit par Dennis Ritchie, le créateur de C, et Brian Kernighan. Attention cependant - c'est ancien et il en contient
+des inexactitudes (enfin, des idées qui ne sont plus considérées comme bonnes) ou des pratiques désormais modifiées.
```
Comment (vendethiel) [inline fr-fr/c-fr.html.markdown:851]: alors pourquoi est-ce *le* livre ?

je suppose encore que c'est dans le fichier de base.

**Review by vendethiel — CHANGES_REQUESTED** [review]

**Comment by verhovsky** [comment]
@vendethiel can you make the edits yourself and merge this please? You can do it from the browser if you click here

editfile

**Comment by vendethiel** [comment]
I think parts of this may have been machine translated. Fixed what I could and merged!

**Pull request merged**